src/test/kotlin/com/gamingbacklog/api/gamingbacklogapi/unit/services/UserServiceTests.kt

```diff
@@ -0,0 +1,157 @@
+    given(libraryService.delete(any())).willAnswer { }
+  }
+
+  fun mockGetSingleLibrary(libraryName: String) {
+    given(libraryService.getSingle(libraryId)).willReturn(Library(id = libraryId, name = libraryName, games = ArrayList()))
+  }
+
+
+
+  @BeforeEach
+  fun initialize() {
+    given(libraryService.convertLibraryToResponse(any())).willCallRealMethod()
+  }
+
+  @AfterEach
+  fun tearDown() {
+    mockUserDb.clear()
+  }
+
+  @Nested
+  @DisplayName("Tests for createUser")
+  inner class CreateUser {
+    // TODO: add more tests in GB-79
+
+    @Test
+    fun shouldCreateNewUser() {
+      val user = User("1772a862dcb22c5d5356b5ec", "AwesomeUser", "xxx", "user@user.com", arrayListOf(libraryId))
+      mockCreateLibrary("All Games")
+      mockSave(user)
+      val result = userService.create(UserRequest("AwesomeUser", "xxx", "user@user.com"))
+      assertNotNull(result)
+      assertEquals(user.displayName, result?.displayName)
+      assertEquals(user.email, result?.email)
+      assertEquals(user.password, result?.password)
+      assertEquals(user.libraries, result?.libraries)
+    }
+
+  }
+
+  @Nested
+  @DisplayName("Tests for createUserLibrary")
+  inner class CreateUserLibrary {
+
+    @Test
+    fun shouldHaveNullUser() {
+      val user = User("1772a862dcb22c5d5356b5ec", "userName", "123", "test@test.com")
+      mockCreateLibrary("Backlog")
+      mockGetOne(user.id)
+      val result = userService.createUserLibrary(user.id, LibraryRequest("Backlog", ArrayList()))
+      assertNull(result)
+    }
+
+    @Test
+    fun shouldSuccessfullyCreateUserLibrary() {
+      val user = User("1772a862dcb22c5d5356b5ec", "userName", "123", "test@test.com")
+      mockUserDb[user.id] = user
+      mockSave(user)
+      mockGetOne(user.id)
+      mockCreateLibrary("Backlog")
+      mockGetSingleLibrary("Backlog")
+      val result = userService.createUserLibrary(user.id, LibraryRequest("Backlog", ArrayList()))
+      assertNotNull(result)
+      assertEquals(mockUserDb[user.id]?.libraries?.get(0), result?.libraries?.get(0)?.id)
+    }
+
+  }
+
+  @Nested
+  @DisplayName("Tests for createUserLibrary")
+  inner class DeleteUserLibrary {
+    @Test
+    fun shouldHaveNullUser() {
+      val user = User("1772a862dcb22c5d5356b5ec", "userName", "123", "test@test.com")
+      mockGetOne(user.id)
+      val result = userService.deleteUserLibrary(user.id, libraryId)
+      assertNull(result)
+    }
+
+    @Test
+    fun shouldHaveSuccessfullyDeletedLibraryFromUser() {
+      val user = User("1772a862dcb22c5d5356b5ec", "userName", "123", "test@test.com", arrayListOf(libraryId))
+      mockUserDb[user.id] = user
+      mockGetOne(user.id)
+      mockGetSingleLibrary("Backlog")
+      mockDeleteLibrary()
+      mockSave(user)
+      val result = userService.deleteUserLibrary(user.id, libraryId)
+      assertNotNull(result)
+      assertEquals(ArrayList<LibraryResponse>(), result?.libraries)
+    }
+
+    @Test
+    fun shouldDeleteNotFoundLibrary() {
```

Comment:
What is this actually testing? When we don't find the library, we're still trying to delete it?

Reply:
We need better error handling throughout tbh. This is basically to show that there is no error when we attempt to delete a library not found. I'm going to remove this and save these edge cases for when we have better error handling (we love pushing things back to this long-awaited error handling ticket 🙃  )